File: queues/parenthesisChecker.java
package queues;

import java.util.Scanner;
import java.util.Stack;

public class parenthesisChecker {
    Scanner scnr = new Scanner(System.in);

    String thisString = scnr.nextLine();

    Stack<Character> myStack = new Stack<>();

    for(
    int i = 0;i<thisString.length();i++)
    {
        if (thisString.charAt(i) == '(') {
            myStack.push('(');
        } else if (thisString.charAt(i) == '[') {
            myStack.push('[');
        } else if (thisString.charAt(i) == '{') {
            myStack.push('{');
        }

        if (thisString.charAt(i) == ')') {
            if (myStack.peek() == ')') {
                myStack.pop();
            } else {
                System.out.println("Not balanced");
            }
        } else if (thisString.charAt(i) == ']') {
            if (myStack.peek() == ']') {
                myStack.pop();
            } else {
                System.out.println("Not balanced");
            }
        } else if (thisString.charAt(i) == '}') {
            if (myStack.peek() == '}') {
                myStack.pop();
            } else {
                System.out.println("Not balanced");
            }
        }
    }
}

File: queues/ParenthesisChecker.py
my_dict: dict[str, str] = {
    "(": ")",
    "{": "}",
    "[": "]",
}

my_string: str = input()
stack: list[str] = []
balanced: bool = True

for ch in my_string:
    if ch in my_dict:
        stack.append(ch)
    elif ch in my_dict.values():
        if not stack:
            balanced = False
            break

        top = stack.pop()
        if my_dict[top] != ch:
            balanced = False
            break

if stack:
    balanced = False

print("Balanced" if balanced else "Not balanced")